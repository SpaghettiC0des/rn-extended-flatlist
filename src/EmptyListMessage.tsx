import React, {FC} from 'react';
import {View, StyleSheet, Text} from 'react-native';
import {EmptyListMessageProps} from './types';

export const DEFAULT_EMPTY_LIST_MESSAGE = 'Empty List';

export const EmptyListMessage: FC<EmptyListMessageProps> = ({
  message = DEFAULT_EMPTY_LIST_MESSAGE,
  textStyle,
  containerStyle,
}) => (
  <View style={[styles.defaultContainer, containerStyle]}>
    <Text
      style={[styles.defaultText, textStyle]}
      accessible
      accessibilityLabel={message}>
      {message}
    </Text>
  </View>
);

const styles = StyleSheet.create({
  defaultContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  defaultText: {
    fontSize: 16,
  },
});
