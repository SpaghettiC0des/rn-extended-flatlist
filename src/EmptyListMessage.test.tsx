import React from 'react';
import {render, RenderAPI} from 'react-native-testing-library';
import {EmptyListMessage, DEFAULT_EMPTY_LIST_MESSAGE} from './EmptyListMessage';

describe('<EmptyListMessage />', () => {
  let rendered: RenderAPI;

  beforeEach(() => {
    rendered = render(<EmptyListMessage />);
  });

  it('should show default message Empty List', () => {
    const {getByA11yLabel} = rendered;

    const actual = getByA11yLabel(DEFAULT_EMPTY_LIST_MESSAGE);

    expect(actual).not.toBeNull();
  });

  it('should accept custom message', () => {
    const {getByA11yLabel, rerender} = rendered;
    rerender(<EmptyListMessage message="Nothing to show" />);

    const actual = getByA11yLabel('Nothing to show');

    expect(actual).not.toBeNull();
  });

  it('should accept custom container style', () => {
    const {toJSON, rerender} = rendered;

    rerender(
      <EmptyListMessage
        containerStyle={{
          backgroundColor: 'red',
        }}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should accept custom text style', () => {
    const {toJSON, rerender} = rendered;

    rerender(
      <EmptyListMessage
        textStyle={{
          color: 'yellow',
        }}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
