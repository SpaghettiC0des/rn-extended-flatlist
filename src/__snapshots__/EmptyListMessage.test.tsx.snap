// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyListMessage /> should accept custom container style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Text
    accessibilityLabel="Empty List"
    accessible={true}
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Empty List
  </Text>
</View>
`;

exports[`<EmptyListMessage /> should accept custom text style 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <Text
    accessibilityLabel="Empty List"
    accessible={true}
    style={
      Array [
        Object {
          "fontSize": 16,
        },
        Object {
          "color": "yellow",
        },
      ]
    }
  >
    Empty List
  </Text>
</View>
`;
